src/Extensions/Features/src/FeatureCollection.cs

```diff
@@ -117,6 +117,15 @@ IEnumerator IEnumerable.GetEnumerator()
     /// <inheritdoc />
     public TFeature? Get<TFeature>()
     {
+        if (typeof(TFeature).IsValueType)
+        {
+            var feature = this[typeof(TFeature)];
+            if (feature is null && Nullable.GetUnderlyingType(typeof(TFeature)) == null)
+            {
+                throw new InvalidOperationException();
```

Comment:
`KeyNotFoundException`?

Reply:
Either that or an error message.

Reply:
Error message added, and incorporated James' musings into the message.